### DISCUSSION TITLE
Suggestions to improve ros2_documentation

### DISCUSSION THREAD

**Discussion by Yadunund** [osrf/ros2_test_cases#1047] [General]
Install options in doc are inconsistently named: https://github.com/osrf/ros2_test_cases/issues/1046

Cc @cdhabecker

**Comment by Yadunund** [comment]
Feedback on colcon sections: https://github.com/osrf/ros2_test_cases/issues/947#issuecomment-1534080571

**Reply by cdhabecker** [reply]
Now that I've started [Creating Your First ROS2 Package](https://github.com/osrf/ros2_test_cases/issues/954), it is clear that there is a copy in the doc. Copies are either perfect or wrong (usually wrong). Therefore:
- Delete the "Create your own package" section of https://docs.ros.org/en/iron/Tutorials/Beginner-Client-Libraries/Colcon-Tutorial.html. Refer to https://docs.ros.org/en/iron/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html, instead.

**Reply by cdhabecker** [reply]
Bigger picture: there is already a progression of tutorials, so let the user go through the progression. There is no good reason to interrupt or end tutorial 1 with a copied chunk of tutorial 2 -- just refer to tutorial 2.

**Comment by Yadunund** [comment]
Incorrect links in Ubuntu-Install-Binary: https://github.com/osrf/ros2_test_cases/issues/722#issuecomment-1533895082

**Comment by mjcarroll** [comment]
Update Windows for correct openssl version https://github.com/ros2/ros2_documentation/pull/3511

**Comment by cdhabecker** [comment]
Feedback on Creating a workspace: https://github.com/osrf/ros2_test_cases/issues/952#issuecomment-1535055826

**Comment by cdhabecker** [comment]
**Ubuntu non-Debian install and rosdep: seems like we're missing an explanation.**

At first glance, the ROS 2 install options appear to have these characteristics:

1. Ubuntu(Debian): Single installation of each ROS version on the computer, location as per APT. For example, /opt/ros/humble and /opt/ros/rolling.
2. Ubuntu(archive or source): Any number of installations of ROS on the computer, location as per user choice. For example, /home/ros2_humble, /home/ros2_iron_1, home/ros2_iron_2, home/ros2_iron_whatever.

But then the instructions for Ubuntu(archive or source) use rosdep to install dependencies, and it appears that rosdep dependencies are specifically APT packages. Now, that might make some sense if the dependencies are strictly ROS-agnostic, e.g., like python3. In that case, each of the ROS installations would share the single set of ROS-agnostic dependencies. But rosdep does not appear to constrain itself to ROS-agnostic dependencies. For example, in my Ubuntu(archive) installation:

```
~/ros2_iron/ros2-linux$ printenv | grep ROS
ROS_VERSION=2
ROS_PYTHON_VERSION=3
ROS_AUTOMATIC_DISCOVERY_RANGE=SUBNET
ROS_DISTRO=rolling

~/ros2_iron/ros2-linux$ echo $AMENT_PREFIX_PATH
/home/habecker/ros2_iron/ros2-linux

~/ros2_iron/ros2-linux$ rosdep check --from-paths share
System dependencies have not been satisfied:
apt	ros-rolling-ament-cmake-ros
apt	ros-rolling-lifecycle-msgs
apt	ros-rolling-rcl
apt	ros-rolling-rclcpp
...
apt	ros-rolling-demo-nodes-py
ERROR[test_ros2trace]: Cannot locate rosdep definition for [test_tracetools]
	rosdep key : test_tracetools
	OS name    : ubuntu
	OS version : jammy
	Data:
<no data>
ERROR[test_tracetools_launch]: Cannot locate rosdep definition for [test_tracetools]
	rosdep key : test_tracetools
	OS name    : ubuntu
	OS version : jammy
	Data:
<no data>
```
What that output tells me is that rosdep does not recognize that the current ROS workspace is a rolling installation that contains its own rolling dependencies. 

I ran "rosdep install --from-paths share", and it refused only because it didn't like the test_ros2trace and test_tracetools_launch packages; otherwise, rosdep seemed ready to use APT to install the ros-rolling-* packages in /opt/ros/rolling. Which should be moot because my ~/ros2_iron/ros2-linux installation already contains those packages.

**Comment by Yadunund** [comment]
Colcon underlay issue when following action tutorials 
https://github.com/ros2/ros2_documentation/issues/3531

**Comment by Yadunund** [comment]
Feedback on package description: https://github.com/osrf/ros2_test_cases/issues/954
